feat/gene/gene_test.go

```diff
@@ -50,24 +50,60 @@ func (c Chr) Name() string           { return string(c) }
 func (c Chr) Description() string    { return "chrom" }
 func (c Chr) Location() feat.Feature { return nil }
 
+// ori implements feat.Feature and is used as location for test objects.
+type ori struct {
+	start, end int
+	name       string
+	desc       string
+	loc        feat.Feature
+	orient     feat.Orientation
+}
+
+func (o ori) Name() string                  { return o.name }
+func (o ori) Description() string           { return o.desc }
+func (o ori) Start() int                    { return o.start }
+func (o ori) End() int                      { return o.end }
+func (o ori) Len() int                      { return o.end - o.start }
+func (o ori) Location() feat.Feature        { return o.loc }
+func (o ori) Orientation() feat.Orientation { return o.orient }
+
 // Define some test objects that will be used in the actual tests
 var (
 	geneA = Gene{
-		ID:     "A",
+		ID:     "geneA",
 		Chrom:  Chr("Y"),
 		Offset: 100,
 		Orient: feat.Forward,
 		Desc:   "forward gene",
 	}
 	geneB = Gene{
-		ID:     "B",
+		ID:     "geneB",
 		Chrom:  Chr("X"),
 		Offset: 100,
 		Orient: feat.Reverse,
 		Desc:   "reverse gene",
 	}
+	geneC = Gene{
```

Comment:
Please make these nested to a degree; the contig should be smaller than the super than the ultra.